From ed0f6d7dc630ed388beb97cb4e9f887717f55179 Mon Sep 17 00:00:00 2001
From: Clayton Smith <argilo@gmail.com>
Date: Sun, 7 Jun 2020 22:04:14 -0400
Subject: [PATCH 33/33] blocks: Fix XMLRPC Client and Server block templates

The XMLRPC Client block is currently broken because it tries to import
the Python 2 XMLRPC module, and because the "addr" and "callback"
parameters are inserted as quoted strings. To fix these problems, I've
switched to the Python 3 version of the XMLRPC module with a fallback to
the Python 2 version, and changed the data types of the problematic
parameters to "raw".

The XMLRPC Server block is currently broken because it tries to import
the Python 2 XMLRPC module. To fix this problem, I've switched to the
Python 3 version of the XMLRPC module with a fallback to the Python 2
version.
---
 gr-blocks/grc/xmlrpc_client.block.yml | 12 ++++++++----
 gr-blocks/grc/xmlrpc_server.block.yml |  7 +++++--
 2 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/gr-blocks/grc/xmlrpc_client.block.yml b/gr-blocks/grc/xmlrpc_client.block.yml
index 308629a680..585b520be5 100644
--- a/gr-blocks/grc/xmlrpc_client.block.yml
+++ b/gr-blocks/grc/xmlrpc_client.block.yml
@@ -5,7 +5,7 @@ flags: [ python ]
 parameters:
 -   id: addr
     label: Address
-    dtype: string
+    dtype: raw
     default: localhost
 -   id: port
     label: Port
@@ -13,15 +13,19 @@ parameters:
     default: '8080'
 -   id: callback
     label: Callback
-    dtype: string
+    dtype: raw
     default: set_
 -   id: variable
     label: Variable
     dtype: raw
 
 templates:
-    imports: import xmlrpclib
-    make: xmlrpclib.Server('http://${addr}:${port}')
+    imports: |-
+        try:
+            from xmlrpc.client import ServerProxy
+        except ImportError:
+            from xmlrpclib import ServerProxy
+    make: ServerProxy('http://${addr}:${port}')
     callbacks:
     - ${callback}(${variable})
 
diff --git a/gr-blocks/grc/xmlrpc_server.block.yml b/gr-blocks/grc/xmlrpc_server.block.yml
index 4328f16f29..f986005354 100644
--- a/gr-blocks/grc/xmlrpc_server.block.yml
+++ b/gr-blocks/grc/xmlrpc_server.block.yml
@@ -14,10 +14,13 @@ parameters:
 
 templates:
     imports: |-
-        import SimpleXMLRPCServer
+        try:
+            from xmlrpc.server import SimpleXMLRPCServer
+        except ImportError:
+            from SimpleXMLRPCServer import SimpleXMLRPCServer
         import threading
     make: |-
-        SimpleXMLRPCServer.SimpleXMLRPCServer((${addr}, ${port}), allow_none=True)
+        SimpleXMLRPCServer((${addr}, ${port}), allow_none=True)
         self.${id}.register_instance(self)
         self.${id}_thread = threading.Thread(target=self.${id}.serve_forever)
         self.${id}_thread.daemon = True
-- 
2.20.1

